src/open_inwoner/kvk/tests/test_views.py

```diff
@@ -166,10 +166,20 @@ def test_post_branches_page_with_incorrect_vestigingsnummer(
         )
 
         doc = PyQuery(response.content)
-        branch_inputs = doc.find("[name='branch_number']")
 
-        # check that pseudo-branch representing company as a whole has been added
-        self.assertEqual(len(branch_inputs), 3)
+        # Verify JSON data structure embedded in script tag for React component
+        # React reads branch data from this JSON rather than parsing DOM elements
+        branch_data_script = doc.find("script.branch-data")
+        self.assertEqual(len(branch_data_script), 1)
+
+        # Parse the JSON data to verify it contains the expected branches
+        import json
+
+        branch_data = json.loads(branch_data_script.text())
+        branch_items = branch_data.get("items", [])
+
+        # Should have 3 items: entire company + 2 branches
+        self.assertEqual(len(branch_items), 3)
```

Comment:
This is a test so we're in control of the conditions: we should know whether we expect there to be a valid object or not, e.g.:

```suggestion
        # Should have 3 items: entire company + 2 branches
        self.assertEqual(len(branch_data["items"]), 3)
```

Reply:
Can be handled in sub-task: https://taiga.maykinmedia.nl/project/open-inwoner/task/3490